website/integrations/services/mastodon/index.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Mastodon
+---
+
+<span class="badge badge--secondary">Support level: Community</span>
+
+## What is Service Name
+
+From https://joinmastodon.org/
+:::note
+Mastodon is free and open-source software for running self-hosted social networking services. It has microblogging features similar to Twitter
+:::
+
+## Preparation
+
+The following placeholders will be used:
+
+-   `inventory.company` is the FQDN of the mastodon install.
+-   `authentik.company` is the FQDN of the authentik install.
+
+## authentik Configuration
+
+### Step 1 - OAuth2/OpenID Provider
+
+In authentik, create a OAuth2/OpenID Provider (under _Applications/Providers_) with these settings :
+
+-   Name : mastodon
+-   Redirect URI: `https://inventory.company/auth/auth/openid_connect/callback`
+
+### Step 3 - Application
+
+In authentik, create an application (under _Resources/Applications_) with these settings :
+
+-   Name: Mastodon
+-   Slug: mastodon
+-   Provider: mastodon
+
+## Mastodon Setup
+
+Configure Mastodon `OIDC_` settings by editing the `.env.production`
+
+Add the following
+
+```
+OIDC_ENABLED=true
+OIDC_DISPLAY_NAME=authentik
+OIDC_DISCOVERY=true
+OIDC_ISSUER=< OpenID Configuration Issuer>
+OIDC_AUTH_ENDPOINT=https://authentik.company/application/o/authorize/
+OIDC_SCOPE=openid,profile,email
+OIDC_UID_FIELD=preferred_username
```

Comment:
I don't know if this field is also used as a username in mastodon, if it isn't it makes more sense to use `sub` here as the username in authentik might change

Reply:
Can you explain in more detail what you mean by " to use `sub `here"?

Reply:
the `sub` field in the OIDC Token is a unique identifier that is (by default) generated based on the User ID and the hashed secret key, which unlike the user's username cannot change

with the current setting, if a user changes their username, they would get a new user in mastodon